### ISSUE TITLE
New Rule - No duplicate destructured exports

### ISSUE THREAD

**Issue by nickofthyme** [benmosher/eslint-plugin-import#1795]
It would be nice to have a rule that combines/simplifies a duplicated object export destructuring. Similar to [`no-duplicates`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-duplicates.md).

This export style might be pretty rare but I think this rule would still be very useful.

### Bad
```ts
export const { defaultColor } = module;
export const { transparentColor } = module;
export const { defaultD3Color } = module;
```

### Good
```ts
export const { defaultColor, transparentColor, defaultD3Color } = module;
```

**Comment by ljharb** [comment]
I'd say it's so rare that i've never once seen it; where does `module` come from in this case?

**Comment by nickofthyme** [comment]
Haha, yeah I know it's rare. The module in my case would come from something like...

```ts
const module = jest.requireActual('../some_file.ts');`
```

Where I then want to `export` only certain functions or properties from that module.

Maybe this would be suitable for simplifying the destructuring itself with or without the `export`. In which case `import` is probably not the best plugin.

**Comment by ljharb** [comment]
Is there a use case outside of jest mocking?

**Comment by nickofthyme** [comment]
None that I can think of outside of complex `index` files that need to use then export things within a module. But again, it's rare. I'm just gonna close this and see about adding a no duplicate destructure rule elsewhere.
